### DISCUSSION TITLE
buildcmake

### DISCUSSION THREAD

**Discussion by giscus** [brainboxdotcc/dpp-comments#19] [General]
# buildcmake

A lightweight C++ Discord API library supporting the entire Discord API, including Slash Commands, Voice/Audio, Sharding, Clustering and more!

https://dpp.dev/buildcmake.html

**Comment by MAGNUSGAO** [comment]
If you see the an error of
 clang: error: unknown argument: '-fcoroutines'"

you might want to turn off the DPP_CORO option in CMake.

**Comment by LorenDB** [comment]
You could alternatively use something like this in your CMakeLists.txt:

``` cmake
find_package(dpp 10.0.23)
if(NOT dpp_FOUND)
    message("Fetching D++ from GitHub...")

    include(FetchContent)
    FetchContent_Declare(dpp
        GIT_REPOSITORY https://github.com/brainboxdotcc/DPP
        GIT_TAG v10.0.23
    )
    FetchContent_MakeAvailable(dpp)
endif()
```

which would let you eliminate (a) the subdirectory and (b) the call to `target_include_directory`.
